staging/src/k8s.io/apiserver/pkg/server/signal.go

```diff
@@ -26,7 +26,7 @@ var onlyOneSignalHandler = make(chan struct{})
 // SetupSignalHandler registered for SIGTERM and SIGINT. A stop channel is returned
 // which is closed on one of these signals. If a second signal is caught, the program
 // is terminated with exit code 1.
-func SetupSignalHandler() (stopCh <-chan struct{}) {
+func SetupSignalHandler() (stopCh chan struct{}) {
```

Comment:
Why the signature change? Receive only is preferred for stop channels

Reply:
Because, this channel is passed to leaderelection, if it is unable to renew, then it should also able to notify the running routines(here does this by closing the channel).

Reply:
then that is a sign you need a second channel. it isn't appropriate for two independent parts of the code to try to close the same channel... that leads to panics.

Reply:
ok, I have thought of that issue. Then I have to find a more appropriate way to do that.

Reply:
done